src/Features/LanguageServer/Protocol/Handler/InlineCompletions/InlineCompletionsHandler.XmlSnippetParsing.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.InlineCompletions;
+
+internal partial class InlineCompletionsHandler
```

Comment:
code here was extracted from similar editor side code that reads snippet xml files and modified a bit to support our use case.

If we need to support XML based snippets long term, it would be nice if an API existed on the editor side to retrieve these.  We would have to still do some conversion since some snippets are context dependent, but reading and parsing the XML could be left to the editor.

That being said, semantic snippets will probably make this irrelevant for built in c# snippets, and the editor is re-thinking the snippet experience so I don't believe it's worth doing anything right now.